merkle/log_verifier.go

```diff
@@ -212,3 +213,35 @@ func (v LogVerifier) VerifyConsistencyProof(snapshot1, snapshot2 int64, root1, r
 
 	return nil // Proof OK.
 }
+
+// VerifiedPrefixHashFromInclusionProof calculates a root hash over leaves
+// [0..index), based on the inclusion |proof| and |leafHash| for a leaf at the
+// specified |index| in a tree of the specified |size| with the passed in
+// |root| hash.
+// Returns an error if the |proof| verification fails. The resulting smaller
+// tree's root hash is trusted iff the bigger tree's |root| hash is trusted.
+func (v LogVerifier) VerifiedPrefixHashFromInclusionProof(
+	index, size int64,
```

Comment:
Would it be clearer to talk always in terms of sizes rather than size and index since we're really dealing with a smaller tree being a prefix of a larger tree?